avocado/utils/pmem.py

```diff
@@ -118,6 +124,69 @@ def run_daxctl_list(self, options=''):
         cmd = '%s list %s' % (self.daxctl, options)
         return json.loads(process.system_output(cmd))
 
+    def dax_online_memory(self, device, region='', no_movable=False):
+        """Online memory from a given devdax device
+
+        :param device: Device from which memory is to be online
+        :param region: Optionally filter device by region
+        :param no_movable: Optionally make the memory non-movable
+
+        :return: True if command succeeds
+        :rtype: bool
+        :raise: :class:`PMemException`, if command fails.
+        """
+        cmd = '%s online-memory %s' % (self.daxctl, device)
+        if region:
+            cmd += ' -r %s' % region
+        if no_movable:
+            cmd += ' --no-movable'
+        if process.system(cmd, shell=True, ignore_status=True):
+            raise PMemException("Failed to online memory with %s" % device)
+        return True
+
+    def dax_offline_memory(self, device, region=''):
+        """Offline memory from a given devdax device
+
+        :param device: Device from which memory is to be offline
+        :param region: Optionally filter device by region
+
+        :return: True if command succeeds
+        :rtype: bool
+        :raise: :class:`PMemException`, if command fails.
+        """
+        cmd = '%s offline-memory %s' % (self.daxctl, device)
+        if region:
+            cmd += ' -r %s' % region
+        if process.system(cmd, shell=True, ignore_status=True):
+            raise PMemException("Failed to offline memory with %s" % device)
+        return True
+
+    def dax_reconfigure_device(self, device, mode='devdax', region='',
+                               no_online=False, no_movable=False):
+        """Reconfigure devdax device into devdax or system-ram mode
+
+        :param device: Device from which memory is to be online
+        :param mode: Mode with which device is to be configured, default:devdax
+        :param region: Optionally filter device by region
+        :param no_online: Optionally don't online the memory(only system-ram)
+        :param no_movable: Optionally mark memory non-movable(only system-ram)
+
+        :return: Property of configured device
+        :rtype: str
+        :raise: :class:`PMemException`, if command fails.
+        """
+        cmd = '%s reconfigure-device %s -m %s' % (self.daxctl, device, mode)
+        if region:
+            cmd += ' -r %s' % region
+        if no_online:
+            cmd += ' -N'
+        if no_movable:
+            cmd += ' --no-movable'
+        prop = process.run(cmd, shell=True, ignore_status=True)
```

Comment:
What is prop? property? I would prefer something more explicit.

Reply:
yes, It is mentioned as a return value in docstring. Does the variable name have to be explicit as well?

Reply:
yes, please, let's avoid abbreviations on the variables. The code is more readable and easy.